sinks/kafka/kafka.go

```diff
@@ -277,6 +296,36 @@ func (k *KafkaSpanSink) Start(cl *trace.Client) error {
 // the bytes, messages and interval settings to your tastes!
 func (k *KafkaSpanSink) Ingest(span *ssf.SSFSpan) error {
 
+	// if we've set a sampleTag, we need to determine whether a span should be
+	// sampled.
+	if k.sampleTag != "" {
+		var hashKey uint32
+		sampleTag, exists := span.Tags[k.sampleTag]
+		if !exists {
+			// If the span isn't tagged appropriately, we should drop it, regardless
+			// of our sample rate.
+			k.logger.Debug("Rejected span without appropriate tag")
+			return nil
+		} else {
+			// Lifted from https://github.com/stathat/consistent/blob/75142be0209ec69bb014c7a1ac7d1a3c892c6424/consistent.go#L238-L245:
+			// if the sample tag value that we're hashing is shorter than 64 bytes, we
+			// need to pad it with zeroes for the crc32.ChecksumIEEE function.
+			if len(sampleTag) < 64 {
+				var scratch [64]byte
+				copy(scratch[:], sampleTag)
+				hashKey = crc32.ChecksumIEEE(scratch[:len(sampleTag)])
+			} else {
+				hashKey = crc32.ChecksumIEEE([]byte(sampleTag))
+			}
+			// Reject any spans whose hash keys end up greater than the threshold that
+			// we previously computed.
+			if hashKey > k.sampleThreshold {
+				k.logger.WithField("hashKey", hashKey).WithField("sampleThreshold", k.sampleThreshold).Debug("Rejected span with appropriate tag based off of sampling rules")
```

Comment:
Since this is unlikely to ever be used on anything of any real volume — that is, mostly in testing — I think we could add more fields to this log line. Adding the tag name and the tag value would probably really help the operator?

Reply:
Sure! I had added these for debugging purposes when writing the code, and in all honesty I was intending to remove them before merging, but I can add more fields to this as well.